backend/fregepoc/indexers/models.py

```diff
@@ -60,3 +61,62 @@ def __iter__(self):
     class Meta:
         verbose_name = _("Github Indexer")
         verbose_name_plural = verbose_name
+
+
+class BitbucketIndexer(BaseIndexer):
+    min_forks = models.PositiveIntegerField(
+        _("min forks"),
+        default=1,
+    )
+
+    current_date = models.DateTimeField(
+        _("current date"),
+        default=bitbucket.DEFAULT_DATE,
+        help_text=_(
+            "The creation date of repository from which to start searching. "
+            "Please note that Bitbucket API paginates repos by creation date, "
+            "so the dates are used to iterate over repositories."
+        ),
+    )
+
+    def __iter__(self):
+        while True:
+            repository_data, next_date = bitbucket.get_next_page(
+                self.current_date
+            )
+
+            if not repository_data:
+                self.rate_limit_exceeded = True
+                break
+
+            self.current_date = next_date
+            self.save(update_fields=["current_date"])
+
+            if (
+                self.min_forks
+                and bitbucket.get_forks_count(repository_data) < self.min_forks
+            ):
+                continue
+
+            clone_url = bitbucket.get_clone_url(repository_data)
+            repo_url = bitbucket.get_repo_url(repository_data)
+            commit_hash = bitbucket.get_last_commit_hash(repository_data)
+
+            if not (clone_url and repo_url and commit_hash):
+                continue
+
+            repo_to_process = Repository(
+                name=repository_data.get("name"),
+                description=repository_data.get("description"),
+                git_url=clone_url,
+                repo_url=repo_url,
+                commit_hash=commit_hash,
+            )
+
+            Repository.objects.bulk_create([repo_to_process])
```

Comment:
Using `bulk_create` to create a single object is a bit of an overkill - `Repository.objects.create` should suffice in this case.